File: JAVA/8. String to Integer (atoi).java
class Solution {
    public int myAtoi(String s) {
        int i = 0, k = 1;
        long d = 0;
        while (i < s.length() && s.charAt(i) == ' ')
            i++;
        if (i < s.length() && s.charAt(i) == '-') {
            k = -1;
            i++;
        } else if (i < s.length() && s.charAt(i) == '+')
            i++;
        while (i < s.length() && s.charAt(i) >= '0' && s.charAt(i) <= '9') {
            d = d * 10 + (s.charAt(i++) - '0');
            if (d * k > Integer.MAX_VALUE)
                return Integer.MAX_VALUE;
            if (d * k < Integer.MIN_VALUE)
                return Integer.MIN_VALUE;
        }
        return (int) (d * k);
    }
}

File: C++/8. String to Integer (atoi).cpp
class Solution {
public:
    int myAtoi(string s) {
        int i=0,k=1;
        long long int d=0;
        while(s[i]==' ')
        i++;
        if(s[i]=='-'){
        k=-1;
        i++;}
        else if(s[i]=='+'){
        k=1;i++;}
        while(s[i]>='0'&&s[i]<='9'){
        d=d*10+(s[i++]-'0');
        if(d*k>INT_MAX)
        return INT_MAX;
        if(d*k<INT_MIN)
        return INT_MIN;
        }
        return d*k;
    }
};
